sympy/geometry/tests/test_polygon_second_moment_of_area.py

```diff
@@ -0,0 +1,155 @@
+    i4=integrate(eq4_y*(x**2),(x,x1,x4))
+
+    i_cal=simplify(i4-i1-i2-i3)
+    triangle=Polygon(p1,p2,p3,p4)
+    i_for=triangle.second_moment_of_area(p)[1]
+    i = int(expand(i_cal-i_for))
+    assert i is 0
+    #I_xx
+    eq1_x=(((p1.x-p2.x)/(p1.y-p2.y))*(y-p1.y))+p1.x
+    eq2_x=(((p2.x-p3.x)/(p2.y-p3.y))*(y-p2.y))+p2.x
+    eq3_x=(((p3.x-p4.x)/(p3.y-p4.y))*(y-p3.y))+p3.x
+    eq4_x=(((p4.x-p1.x)/(p4.y-p1.y))*(y-p4.y))+p4.x
+
+    i1=integrate(eq1_x*(y**2),(y,y1,y2))
+    i2=integrate(eq2_x*(y**2),(y,y2,y3))
+    i3=integrate(eq3_x*(y**2),(y,y3,y4))
+    i4=integrate(eq4_x*(y**2),(y,y1,y4))
+
+    i_cal=simplify(-i4+i1+i2+i3)
+    triangle=Polygon(p1,p2,p3,p4)
+    i_for=triangle.second_moment_of_area(p)[0]
+    i = int(expand(i_cal-i_for))
+    assert i is 0
+
+    #Pentagon (eqation of side of pentagon) I_yy
+    eq1_y=(((p1.y-p2.y)/(p1.x-p2.x))*(x-p1.x))+p1.y
+    eq2_y=(((p2.y-p3.y)/(p2.x-p3.x))*(x-p2.x))+p2.y
+    eq3_y=(((p3.y-p4.y)/(p3.x-p4.x))*(x-p3.x))+p3.y
+    eq4_y=(((p4.y-p5.y)/(p4.x-p5.x))*(x-p4.x))+p4.y
+    eq5_y=(((p5.y-p1.y)/(p5.x-p1.x))*(x-p5.x))+p5.y
+
+    i1=integrate(eq1_y*(x**2),(x,x1,x2))
+    i2=integrate(eq2_y*(x**2),(x,x2,x3))
+    i3=integrate(eq3_y*(x**2),(x,x3,x4))
+    i4=integrate(eq4_y*(x**2),(x,x4,x5))
+    i5=integrate(eq5_y*(x**2),(x,x1,x5))
+
+    i_cal = simplify(i5-i1-i2-i3-i4)
+    pentagon=Polygon(p1,p2,p3,p4,p5)
+    i_for = pentagon.second_moment_of_area(p)[1]
+    i = int(expand(i_cal-i_for))
+    assert i is 0
+    #I_xx
+    eq1_x=(((p1.x-p2.x)/(p1.y-p2.y))*(y-p1.y))+p1.x
+    eq2_x=(((p2.x-p3.x)/(p2.y-p3.y))*(y-p2.y))+p2.x
+    eq3_x=(((p3.x-p4.x)/(p3.y-p4.y))*(y-p3.y))+p3.x
+    eq4_x=(((p4.x-p5.x)/(p4.y-p5.y))*(y-p4.y))+p4.x
+    eq5_x=(((p5.x-p1.x)/(p5.y-p1.y))*(y-p5.y))+p5.x
+
+    i1=integrate(eq1_x*(y**2),(y,y1,y2))
+    i2=integrate(eq2_x*(y**2),(y,y2,y3))
+    i3=integrate(eq3_x*(y**2),(y,y3,y4))
+    i4=integrate(eq4_x*(y**2),(y,y4,y5))
+    i5=integrate(eq5_x*(y**2),(y,y1,y5))
+
+    i_cal = simplify(-i5+i1+i2+i3+i4)
+    pentagon=Polygon(p1,p2,p3,p4,p5)
+    i_for = pentagon.second_moment_of_area(p)[0]
+    i = int(expand(i_cal-i_for))
+    assert i is 0
+    #Hexagon (eqation of side of Hexagon) I_yy
+    eq1_y=(((p1.y-p2.y)/(p1.x-p2.x))*(x-p1.x))+p1.y
+    eq2_y=(((p2.y-p3.y)/(p2.x-p3.x))*(x-p2.x))+p2.y
+    eq3_y=(((p3.y-p4.y)/(p3.x-p4.x))*(x-p3.x))+p3.y
+    eq4_y=(((p4.y-p5.y)/(p4.x-p5.x))*(x-p4.x))+p4.y
+    eq5_y=(((p5.y-p6.y)/(p5.x-p6.x))*(x-p5.x))+p5.y
+    eq6_y=(((p6.y-p1.y)/(p6.x-p1.x))*(x-p6.x))+p6.y
+
+    i1=integrate(eq1_y*(x**2),(x,x1,x2))
+    i2=integrate(eq2_y*(x**2),(x,x2,x3))
+    i3=integrate(eq3_y*(x**2),(x,x3,x4))
+    i4=integrate(eq4_y*(x**2),(x,x4,x5))
+    i5=integrate(eq5_y*(x**2),(x,x5,x6))
+    i6=integrate(eq6_y*(x**2),(x,x1,x6))
+
+    i_cal=simplify(i6-i1-i2-i3-i4-i5)
+    hexagon=Polygon(p1,p2,p3,p4,p5,p6)
+    i_for = hexagon.second_moment_of_area(p)[1]
+    i = int(expand(i_for-i_cal))
+    assert i is 0
```

Comment:
Why is this a good assertion? it isn't clear to me if this is a good, thorough test. It may be better to move these to the documentation  as comprehensive examples and use simpler shapes like your so string example for these unit tests.

Reply:
found the second moment of area from fundamental formula (integrate(x**2,dA)).for solving this integration break the integration into different integration(i1,i2,i3,i4,i5 etc).
![img_20171206_143118374](https://user-images.githubusercontent.com/23614977/33806155-a46590a0-dde9-11e7-890b-ce30ea5de865.jpg)
![img_20171206_143127034](https://user-images.githubusercontent.com/23614977/33806176-fd78987c-dde9-11e7-99ab-e0d5e04ae0ca.jpg).how do i break the integration is in the image. found the second moment of area from the implemented function and diffrence beetween both the expression should be zero. It is more genral (yes but difficul to understand).i will add some simple test also.

Reply:
added test for simple shapes.